### PR TITLE
prov/efa: prefix for cq changes in efa

### PR DESCRIPTION
**Pull request by dancejic**

### CODE REVIEW

**Comment by rajachan** [comment]
@dancejic the Travis build failures are real issues.
```
  CC       prov/efa/src/src_libfabric_la-efa_ep.lo

prov/efa/src/efa_ep.c: In function ‘efa_ep_progress_internal’:

prov/efa/src/efa_ep.c:668:2: error: ‘for’ loop initial declarations are only allowed in C99 mode

  for (int i = 0; i < ret; i++) {

  ^

prov/efa/src/efa_ep.c:668:2: note: use option -std=c99 or -std=gnu99 to compile your code

make[1]: *** [prov/efa/src/src_libfabric_la-efa_ep.lo] Error 1

make[1]: *** Waiting for unfinished jobs....

make[1]: Leaving directory `/home/travis/build/ofiwg/libfabric'

make: *** [all] Error 2
```

You'd have to use the config flags from the [.travis.yml](https://github.com/ofiwg/libfabric/blob/master/.travis.yml) file to reproduce them.

**Comment by jsquyres** [comment]
Huh.  I thought Libfabric has long-since required a C99 compiler.

Is that incorrect?

**Comment by dancejic** [comment]
It looks like it doesn't like "int i". After checking the rest of the code, it's consistent to declare i before the loop. I'll fix it and build with the travis config flags and make sure there aren't any other issues.

**Comment by rajachan** [comment]
> Huh. I thought Libfabric has long-since required a C99 compiler.
> Is that incorrect?

I do see `AC_PROG_CC_C99` in configure. The travis environment seems to have noped out of it:

> checking for gcc option to accept ISO C99... unsupported

**Comment by jsquyres** [comment]
But we do stuff like this all over the place:

```c
struct fi_provider core_prov = {
        .name = "core",
        .version = 1,
        .fi_version = FI_VERSION(FI_MAJOR_VERSION, FI_MINOR_VERSION)
};
```

That's C99 syntax.  It is most definitely *not* C89 syntax.

Meaning: I think we _require_ a C99 compiler.  If some compilers sorta do C99 without us asking for anything special (e.g., apparently the Travis compiler -- since it doesn't choke on the above syntax, but it _does_ choke on `for (int i = 3 ...)`, then that's still not good enough.

Ah, here's the commit: 98bd1afd15e886a8b32ce8ed832b5b34dc5d5988

Apparently this is not enough to _require_  a C99 compiler, which is a bit of a surprise to me.  We should fix that (that's outside the scope of this PR, though).

Additionally, I would be very surprised if Travis does not have support for a C99 compiler.

File: prov/efa/src/efa_domain.c [hunk 1]
```diff
@@ -111,7 +112,7 @@ static struct fi_ops_domain efa_domain_ops = {
 	.cq_open = efa_cq_open,
 	.endpoint = efa_ep_open,
 	.scalable_ep = fi_no_scalable_ep,
-	.cntr_open = fi_no_cntr_open,
+	.cntr_open = efa_cntr_open,
```
Comment (rajachan) [inline prov/efa/src/efa_domain.c]: Does the `rdm_cntr_pingpong` test run successfully now? If it does, we should remove it from our fabtest exclude list.

File: prov/efa/src/efa_ep.c [hunk 2]
```diff
@@ -611,6 +633,132 @@ static struct fi_ops efa_ep_ops = {
...
+				     temp_cq_entry->buf,
+				     temp_cq_entry->data,
+				     temp_cq_entry->tag);
+
+		temp_cq_entry = (struct fi_cq_tagged_entry *)
+				((uint8_t *)temp_cq_entry + efa_cq->entry_size);
+	}
+}
+
+void efa_ep_rdm_progress(struct util_ep *ep)
```
Comment (rajachan) [inline prov/efa/src/efa_ep.c]: This and the dgrm progress function are doing the same thing, can we combine them into one and register with ofi_endpoint_init()?

File: prov/efa/src/efa_ep.c [hunk 3]
```diff
@@ -611,6 +633,132 @@ static struct fi_ops efa_ep_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
+static void efa_ep_progress_internal(struct efa_cq *efa_cq, uint64_t flags)
+{
+	struct util_cq *cq = &efa_cq->util_cq;
+	int cq_entries = 500;
```
Comment (rajachan) [inline prov/efa/src/efa_ep.c]: Can we make this a macro and define it up front in the file (and make it a tunable param)?

File: prov/efa/src/efa_domain.c (hunk 1 above)
Comment (dancejic) [reply prov/efa/src/efa_domain.c]: It still doesn't work if I remove if from the exclude list. I think the efa counter would still need to be set up from rxr in order for this test to pass.

**Pull request merged**